### PR TITLE
felix 139&92

### PR DESCRIPTION
**Pull request by phoenixzqy**

### CODE REVIEW

File: 92/felix.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *     constructor(val?: number, next?: ListNode | null) {
+ *         this.val = (val===undefined ? 0 : val)
+ *         this.next = (next===undefined ? null : next)
+ *     }
+ * }
+ */
+
+function reverseBetween(head: ListNode | null, left: number, right: number): ListNode | null {
+    let valArr: { [index: string]: number } = {};
+
```
Comment (t3rryw1) [inline 92/felix.ts:15]: if you are doing value swap... I guess you don't really need recursive. You can iterativelly hash the value and start replacing once it pass half way.

**Review by t3rryw1 — APPROVED** [review]

File: 92/felix.ts (hunk 1 above)
Comment (phoenixzqy) [reply 92/felix.ts:15]: But this is a linked list, not array. You won't be able to easily access all nodes as you need, unless you store them into another variable.
eg: how to swap index 2 and 8 if you are at 4?

File: 92/felix.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+function reverseBetween(head: ListNode | null, left: number, right: number): ListNode | null {
+    let valArr: { [index: string]: number } = {};
+    
+    function rec(node: ListNode | null, index) {
+        if (!node) return;
+        // record nodes that needs to be reversed
+        if (index >= left && index <= right) {
+            valArr[index] = node.val;
```
Comment (t3rryw1) [inline 92/felix.ts:20]: I think we can use a queue to push values in, then pop it out on the other side, sort like a FIFO.

Comment (phoenixzqy) [reply 92/felix.ts:20]: make sense, but there is no queue built-in in js. And talking about memory consumption, there is no difference between saving values into an object and a so called queue (an array in js).

Comment (t3rryw1) [reply 92/felix.ts:20]: haha, sure, it's just makes me feel better.

Comment (t3rryw1) [reply 92/felix.ts:20]: haha, sure, it's just makes me feel better.

**Pull request merged**